docs/source/notebooks_and_ipython/kedro_and_notebooks.md

```diff
@@ -236,6 +236,54 @@ If a Jupyter kernel with the name `kedro_<package_name>` already exists then it
 
 You can use the `jupyter kernelspec` set of commands to manage your Jupyter kernels. For example, to remove a kernel, run `jupyter kernelspec remove <kernel_name>`.
 
+### Debugging with %debug and %pdb
+
+ `%debug` will run post-mortem debugging using IPython. Running `%debug` immediately after an error has occurred will:
+ - Load the stack trace of the last unhandled exception.
+ - Stop the program at the point where the exception occurred.
+ - Open an interactive shell where the user can navigate through the stack trace.
+
+ You can then inspect the value of expressions and arguments, or add breakpoints to the code.
+
+<details>
+<summary>Click to see an example.</summary>
+
+![jupyter_ipython_debug_command](../meta/images/jupyter_ipython_debug_command.gif)
+
+</details>
+
+---
+
+You can set up the deubgger to run automatically when an exception occurs by running `%pdb 1` or `%pdb on` before executing a program. This automatic behaviour can be disabled with `%pdb 0` or `%pdb off`.
+
+<details>
+<summary>Click to see an example.</summary>
+
+ ![jupyter_ipython_pdb_command](../meta/images/jupyter_ipython_pdb_command.gif)
+
+</details>
+
+---
+
+Below are some examples of the possible commands that can be used to interact with the ipdb shell:
+
+| Command           | Description                                           |
+| ----------------- | ----------------------------------------------------- |
+| `list`            | Show the current location in the file                 |
+| `h(elp)`          | Show a list of commands, or find help on a specific command |
+| `q(uit)`          | Quit the debugger and the program                     |
+| `c(ontinue)`      | Quit the debugger, continue in the program             |
+| `n(ext)`          | Go to the next step of the program                     |
+| `<enter>`         | Repeat the previous command                            |
+| `p(rint)`         | Print variables                                       |
+| `s(tep)`          | Step into a subroutine                                |
+| `r(eturn)`        | Return out of a subroutine                            |
+| `b(reak)`         | Insert a breakpoint                                   |
+| `a(rgs)`          | Print the argument list of the current function        |
+
+For more information, use the `help` command in the debugger, or take a look at ipdb's [online documentation](https://github.com/gotcha/ipdb).
```

Comment:
```suggestion
For more information, use the `help` command in the debugger, or take a look at the [ipdb documentation](https://github.com/gotcha/ipdb).
```

I feel a little hesitant to call this documentation, @stichbury any suggestions here?